### PR TITLE
Auth/config in FeatherBB middleware

### PR DESCRIPTION
**Pull request by peloyeje**

### CODE REVIEW

File: include/routes.php [hunk 1]
```diff
@@ -157,10 +157,10 @@
...
     });
-    
+
 });
 
 // 404 not found
-$feather->notFound(function () use ($lang_common) {
-    message($lang_common['Bad request'], '404');
+$feather->notFound(function () use ($feather) {
+    $feather->halt(404, 'Not found');
```
Comment (adaur) [inline include/routes.php]: I'd prefer to use "message" to display an user-friendly page

File: Slim/Extras/Middleware/FeatherBB.php [hunk 2]
```diff
@@ -41,11 +137,243 @@ public function call()
...
+            }
+
+            // Check if current user is banned
+            check_bans();
+
+            // Update online list
+            update_users_online();
+
+            // Configure Slim
+            $this->app->config('templates.path', (is_dir('style/'.$this->app->user->style.'/view')) ? FEATHER_ROOT.'style/'.$this->app->user->style.'/view' : FEATHER_ROOT.'view');
```
Comment (adaur) [inline Slim/Extras/Middleware/FeatherBB.php:393]: Can you please remove the function that did this?

File: Slim/Extras/Middleware/FeatherBB.php [hunk 3]
```diff
@@ -1,36 +1,132 @@
...
+            include $this->data['forum_env']['FORUM_CACHE_DIR'].'cache_config.php';
+        } else {
+            require_once $this->data['forum_env']['FEATHER_ROOT'].'include/cache.php';
+            generate_config_cache();
+            require $this->data['forum_env']['FORUM_CACHE_DIR'].'cache_config.php';
+        }
+        return $feather_config;
+    }
+
+    public static function load_default_settings()
```
Comment (adaur) [inline Slim/Extras/Middleware/FeatherBB.php:52]: I don't understand the purpose of this ?

File: Slim/Extras/Middleware/FeatherBB.php [hunk 4]
```diff
@@ -1,36 +1,132 @@
...
 class FeatherBB extends \Slim\Middleware
 {
-	protected $settings = array();
+    protected $data = array();
 
-	public function __construct()
+	public function __construct(array $user_forum_settings = array())
 	{
+        // Define forum env constants
+        $this->data['forum_env'] = array(
```
Comment (adaur) [inline Slim/Extras/Middleware/FeatherBB.php:24]: Will this override the current constants ?

File: include/routes.php (hunk 1 above)
Comment (peloyeje) [inline include/routes.php]: All right :) It was meant to be a temporary solution, while waiting for a better error page which uses Slim interface

File: Slim/Extras/Middleware/FeatherBB.php (hunk 2 above)
Comment (peloyeje) [inline Slim/Extras/Middleware/FeatherBB.php:393]: Meaning the get_path_view function in functions.php ?

File: Slim/Extras/Middleware/FeatherBB.php (hunk 3 above)
Comment (peloyeje) [inline Slim/Extras/Middleware/FeatherBB.php:52]: This function is called when the middleware merges the current settings with the forum config stored in the DB, and returns the config array : pointless, but it's a wrapper for the cache loading procedure in order for the call function to be cleaner :)

File: Slim/Extras/Middleware/FeatherBB.php (hunk 4 above)
Comment (peloyeje) [inline Slim/Extras/Middleware/FeatherBB.php:24]: No, it only defines the constants which were originally defined in common.php. Keep in mind that the middleware is called before the Slim application in the framework stack

File: Slim/Extras/Middleware/FeatherBB.php (hunk 2 above)
Comment (adaur) [inline Slim/Extras/Middleware/FeatherBB.php:393]: Yup

**Comment by adaur** [comment]
:+1:

**Pull request merged**